lib/smb2/packet.rb

```diff
@@ -174,13 +304,13 @@ def self.data_buffer(name, bit_length=16, opts={})
   def initialize(*args)
     @data_buffers = {}
     super do
+      yield self if block_given?
```

Comment:
You should document the block using `@yield`, `@yieldparam` and `@yieldreturn`.  Since you're not capturing the return value from `yield` you should do `@yieldreturn [void]` to indicate that nothing is done with the return value from the block.